test/distributed/org/apache/cassandra/fuzz/harry/integration/dsl/HistoryBuilderOverridesTest.java

```diff
@@ -0,0 +1,376 @@
+                        try
+                        {
+                            history.forPartition(pdIdx).ensureClustering(new Object[]{ override });
+                            overrides.add(override);
+                        }
+                        catch (IllegalStateException t)
+                        {
+                            // could not override twice
+                        }
+                    }
+
+                    for (int i = 0; i < partitionSize; i++)
+                        history.visitPartition(pdIdx).insert(i);
+
+                    visitor.replayAll();
+                    long visitedPd = history.forPartition(pdIdx).pd();
+                    {
+                        Object[][] res = sut.execute(Query.selectPartition(history.schema(), visitedPd, false).toSelectStatement(),
+                                                     SystemUnderTest.ConsistencyLevel.ALL);
+
+                        int found = 0;
+                        for (int i = 0; i < res.length; i++)
+                        {
+                            Object[] row = res[i];
+                            Object v = row[1];
+                            if (overrides.contains(v))
+                                found++;
+                        }
+                        Assert.assertEquals("Should have mutated exactly one CK", found, overrides.size());
+                    }
+                    history.validateAll(tracker, sut);
+                }
+            }
+        }
+    }
+
+    @Test
+    public void ckOverrideWithDeleteTestSingleColumn()
+    {
+        Supplier<SchemaSpec> supplier = SchemaGenerators.progression(SchemaGenerators.DEFAULT_SWITCH_AFTER);
+
+        int partitionSize = 5;
+        for (int cnt = 0; cnt < SchemaGenerators.DEFAULT_RUNS; cnt++)
+        {
+            SchemaSpec schema = supplier.get();
+            beforeEach();
+
+            DataTracker tracker = new DefaultDataTracker();
+            beforeEach();
+            sut.schemaChange(schema.compile().cql());
+
+            TokenPlacementModel.ReplicationFactor rf = new TokenPlacementModel.SimpleReplicationFactor(1);
+
+            HistoryBuilder history = new HistoryBuilder(seed, partitionSize, 1, schema, rf);
+            ReplayingVisitor visitor = history.visitor(tracker, sut, SystemUnderTest.ConsistencyLevel.ALL);
+
+            EntropySource rng = new JdkRandomEntropySource(seed);
+            for (int i = 0; i < partitionSize; i++)
+            {
+                history.visitPartition(1,
+                                       (e) -> schema.ckGenerator.inflate(rng.next()))
+                       .insert(i);
+            }
+
+            for (int i = 0; i < partitionSize; i++)
+            {
+                history.visitPartition(1)
+                       .deleteRow(i);
+
+                visitor.replayAll();
+                history.validateAll(tracker, sut);
```

Comment:
nit: Can we validate after the loop, or is there something we'd miss if we didn't on every iteration?

Reply:
I thought it might be useful to validate every iteration since some problems may get shadowed by subsequent writes. it should  be relatively cheap still